server/repository/src/db_diesel/assets/asset.rs

```diff
@@ -208,40 +215,16 @@ fn create_filtered_query(filter: Option<AssetFilter>) -> BoxedAssetQuery {
         apply_date_filter!(query, installation_date, asset_dsl::installation_date);
         apply_date_filter!(query, replacement_date, asset_dsl::replacement_date);
 
-        if let Some(category_id) = category_id {
-            let mut sub_query = asset_catalogue_item_dsl::asset_catalogue_item
-                .select(asset_catalogue_item_dsl::id.nullable())
-                .into_boxed();
-            apply_equal_filter!(
-                sub_query,
-                Some(category_id),
-                asset_catalogue_item_dsl::asset_category_id
-            );
-            query = query.filter(asset_dsl::asset_catalogue_item_id.eq_any(sub_query));
-        }
-
-        if let Some(class_id) = class_id {
-            let mut sub_query = asset_catalogue_item_dsl::asset_catalogue_item
-                .select(asset_catalogue_item_dsl::id.nullable())
-                .into_boxed();
-            apply_equal_filter!(
-                sub_query,
-                Some(class_id),
-                asset_catalogue_item_dsl::asset_class_id
-            );
-            query = query.filter(asset_dsl::asset_catalogue_item_id.eq_any(sub_query));
-        }
+        apply_equal_filter!(query, category_id, asset_dsl::asset_category_id);
+        apply_equal_filter!(query, class_id, asset_dsl::asset_class_id);
+        apply_equal_filter!(query, type_id, asset_dsl::asset_type_id);
 
-        if let Some(type_id) = type_id {
-            let mut sub_query = asset_catalogue_item_dsl::asset_catalogue_item
-                .select(asset_catalogue_item_dsl::id.nullable())
-                .into_boxed();
+        if let Some(value) = is_non_catalogue {
             apply_equal_filter!(
-                sub_query,
-                Some(type_id),
-                asset_catalogue_item_dsl::asset_type_id
+                query,
+                Some(EqualFilter::is_null(value)),
+                asset_dsl::asset_catalogue_item_id
```

Comment:
```suggestion
        if let Some(is_non_catalogue) = is_non_catalogue {
            query = match is_non_catalogue {
                true => query.filter(asset_dsl::asset_catalogue_item_id.is_null()),
                false => query.filter(asset_dsl::asset_catalogue_item_id.is_not_null()),
            }
        }
```

I would be tempted to write something like this, but what you've got does work :)